currency/forexprovider/exchangeratesapi.io/exchangeratesapi.go

```diff
@@ -47,34 +63,45 @@ func cleanCurrencies(baseCurrency, symbols string) string {
 		}
 
 		// remove and warn about any unsupported currencies
-		if !strings.Contains(exchangeRatesSupportedCurrencies, x) { // nolint:gocritic
-			log.Warnf(log.Global,
-				"Forex provider ExchangeRatesAPI does not support currency %s, removing from forex rates query.\n", x)
-			continue
+		if len(e.supportedCurrencies) > 0 {
+			if !strings.Contains(strings.Join(e.supportedCurrencies, ","), x) {
+				log.Warnf(log.Global,
+					"Forex provider ExchangeRatesAPI does not support currency %s, removing from forex rates query.\n", x)
+				continue
+			}
 		}
 		cleanedCurrencies = append(cleanedCurrencies, x)
 	}
 	return strings.Join(cleanedCurrencies, ",")
 }
 
+// GetSymbols returns a list of supported symbols
+func (e *ExchangeRates) GetSymbols() (map[string]string, error) {
+	resp := struct {
+		Symbols map[string]string `json:"symbols"`
+	}{}
+	return resp.Symbols, e.SendHTTPRequest("symbols", url.Values{}, &resp)
+}
+
 // GetLatestRates returns a map of forex rates based on the supplied params
 // baseCurrency - USD	[optional] The base currency to use for forex rates, defaults to EUR
 // symbols - AUD,USD	[optional] The symbols to query the forex rates for, default is
 // all supported currencies
-func (e *ExchangeRates) GetLatestRates(baseCurrency, symbols string) (Rates, error) {
+func (e *ExchangeRates) GetLatestRates(baseCurrency, symbols string) (*Rates, error) {
 	vals := url.Values{}
-
-	if len(baseCurrency) > 0 {
+	if len(baseCurrency) > 0 && e.APIKeyLvl <= apiKeyFree && !reflect.DeepEqual("EUR", baseCurrency) {
```

Comment:
I'm unsure why you've used reflect.DeepEqual for a string comparison. Perhaps strings.Equalfold? Or even `!=` if you're not checking case
``` 
BenchmarkDeepEqual-8   	58.34 ns/op
BenchmarkEqualFold-8   	5.040 ns/op
BenchmarkEquals-8   	0.2716 ns/op
```